great_expectations/expectations/row_conditions.py

```diff
@@ -56,6 +61,43 @@ class ConditionParserError(GreatExpectationsError):
     pass
 
 
+class RowConditionParserType(enum.Enum):
+    GE = "ge"  # GE intermediate language
+    SPARK = "spark"
+    SPARK_SQL = "spark_sql"
+    PANDAS = "pandas"
+    PYTHON = "python"
+    SQL = "sql"
+
+
+class RowCondition:
+    """Condition that can be used to filter rows in a data set.
+
+    Attributes:
+        condition: String of the condition
+        type_: Format of the condition e.g. for parsing
+    """
+
+    def __init__(self, condition: str, type_: RowConditionParserType):
+        self._condition = condition
+        self._type_ = type_
+
+    @property
+    def condition(self):
+        return self._condition
+
+    @property
+    def type_(self):
+        return self._type_
+
+    def __eq__(self, other: "RowCondition"):
+        if not isinstance(other, self.__class__):
+            raise GreatExpectationsTypeError(
+                "Only comparison of the same type of object is supported."
+            )
+        return all([self.condition == other.condition, self.type_ == other.type_])
```

Comment:
Could this be a prime candidate for a dataclass? We get `__eq__` for free then right?

Reply:
Actually let me check with @donaldheppner - I think there is a reason I'm forgetting. If not happy to revert.

Reply:
Reverted after discussion - we can always change from a dataclass if we add methods in the future.